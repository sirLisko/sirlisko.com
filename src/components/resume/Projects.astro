---
import type { Resume } from "../../types";

interface Props {
  projects: Resume["projects"];
}
const { projects } = Astro.props;
---

<article class="section projects">
  <h1>Side Projects</h1>
  {
    projects.map((prj) => (
      <div>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr) => (
          <p>{blr}</p>
        ))}
      </div>
    ))
  }
  <div>
    To see a full list of my side projects go to{" "}
    <a href="/projects">https://sirlisko.com/projects</a>.
  </div>
</article>
