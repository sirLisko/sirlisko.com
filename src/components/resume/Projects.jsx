import React from "react";

const Projects = ({ projects }) => (
  <article className="section projects">
    <h1>Side Projects</h1>
    {projects.map(prj => (
      <div key={prj.name}>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
    <div>
      To see a full list of my side projects go to{" "}
      <a href="/projects">https://sirlisko.com/projects</a>.
    </div>
  </article>
);

export default Projects;
