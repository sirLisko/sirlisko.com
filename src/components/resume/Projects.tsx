import React from "react";

interface ProjectProps {
  projects: Array<{
    name: string;
    url?: string;
    blurp: string[];
  }>;
}

const Projects = ({ projects }: ProjectProps) => (
  <article className="section projects">
    <h1>Side Projects</h1>
    {projects.map((prj) => (
      <div key={prj.name}>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
    <div>
      To see a full list of my side projects go to{" "}
      <a href="/projects">https://sirlisko.com/projects</a>.
    </div>
  </article>
);

export default Projects;
