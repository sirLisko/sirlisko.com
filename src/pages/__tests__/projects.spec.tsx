import React from "react";
import { render } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Projects from "../projects";

jest.mock("../../../data/projects.ts", () => [
  {
    title: "Foo",
    description: ["line 1", "line 2"],
    logo: "https://sirlisko.com/logo",
    links: {
      website: "https://foo",
      github: "https://github.com/sirlisko/foo",
    },
    tech: ["foo", "Foo"],
  },
  {
    title: "Bar",
    description: ["line a", "line b"],
    logo: "https://sirlisko.com/logo",
    links: {
      website: "https://bar",
      github: "https://github.com/sirlisko/bar",
    },
    tech: ["bar", "Bar"],
  },
]);

jest.mock("gatsby", () => ({
  useStaticQuery: jest.fn(),
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(({ to, children }) => <a href={to}>{children}</a>),
}));
const mockedStaticQuery = useStaticQuery as jest.Mock;

beforeEach(() => {
  mockedStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: `I am the title`,
      },
    },
  }));
});

describe("Projects Page", () => {
  it("should render properly", () => {
    const { asFragment } = render(<Projects />);
    expect(asFragment()).toMatchSnapshot();
  });
});
