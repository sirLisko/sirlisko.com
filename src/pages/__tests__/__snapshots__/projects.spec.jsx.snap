// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="projects-page"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="sirlisko face"
          src="/images/sirlisko.svg"
        />
      </a>
    </div>
    <section>
      <h1>
        My Side Projects
      </h1>
      <ul>
        <li
          class="project"
        >
          <div
            class="project__main"
          >
            <h3>
              Foo
            </h3>
            <p>
              line 1
            </p>
            <p>
              line 2
            </p>
          </div>
          <img
            alt="Foo logo"
            src="https://sirlisko.com/logo"
          />
          <div
            class="project__details"
          >
            <p
              class="project__details__links"
            >
              <a
                href="https://foo"
              >
                website
              </a>
              <a
                href="https://github.com/sirlisko/foo"
              >
                github
              </a>
            </p>
            <ul
              class="project__details__tech"
            >
              <li>
                foo
              </li>
              <li>
                Foo
              </li>
            </ul>
          </div>
        </li>
        <li
          class="project"
        >
          <div
            class="project__main"
          >
            <h3>
              Bar
            </h3>
            <p>
              line a
            </p>
            <p>
              line b
            </p>
          </div>
          <img
            alt="Bar logo"
            src="https://sirlisko.com/logo"
          />
          <div
            class="project__details"
          >
            <p
              class="project__details__links"
            >
              <a
                href="https://bar"
              >
                website
              </a>
              <a
                href="https://github.com/sirlisko/bar"
              >
                github
              </a>
            </p>
            <ul
              class="project__details__tech"
            >
              <li>
                bar
              </li>
              <li>
                Bar
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </section>
  </div>
</DocumentFragment>
`;
